cynghanedd.py

```diff
@@ -194,9 +194,20 @@ def choose_word(solutions, word):
     remaining_solutions = [ solution[1:] for solution in solutions if solution[0] == skeleton ]
     return remaining_solutions
 
-if __name__ == '__main__':
-    import sys
+def complete_cyng(words, skeleton, solution):
+    while solution and solution[0]:
+        next_words = ' '.join(sorted(first_words(solution)))
+        print(next_words)
+        print("Pick a word (or enter comma to go back a step): ", end="", flush=True)
+        word = sys.stdin.readline().strip()
+        while word == ",":
+            word, words, solution = back_cynghanedd(words, skeleton)
+        words.append(word)
+
+        solution = choose_word(solution, word)
+    return words, solution
```

Comment:
When I have a function that returns something complex (like in this case returning two things), I usually include a comment above the function explaining what it returns.